packages/lib/src/interactions/hooks.ts

```diff
@@ -5,46 +5,67 @@ import { useCallback, useEffect } from 'react';
 import { Vector2, Vector3 } from 'three';
 
 import { useVisCanvasContext } from '../vis/shared/VisCanvasProvider';
-import { getWorldFOV, htmlToWorld } from '../vis/utils';
+import { htmlToWorld } from '../vis/utils';
 import { useInteractionsContext } from './InteractionsProvider';
+import Box from './box';
 import type {
   CanvasEvent,
   CanvasEventCallbacks,
   InteractionEntry,
   ModifierKey,
 } from './models';
-import { clampPositionToArea } from './utils';
 
 const ZOOM_FACTOR = 0.95;
 const ONE_VECTOR = new Vector3(1, 1, 1);
 const MODIFIER_KEYS: ModifierKey[] = ['Alt', 'Control', 'Shift'];
 
 export function useMoveCameraTo() {
-  const { visSize } = useVisCanvasContext();
+  const { canvasSize, visSize } = useVisCanvasContext();
 
   const camera = useThree((state) => state.camera);
   const invalidate = useThree((state) => state.invalidate);
 
   return useCallback(
     (worldPt: Vector3) => {
-      const { position } = camera;
+      const { position, scale } = camera;
 
-      const { topRight, bottomLeft } = getWorldFOV(camera);
-      const width = Math.abs(topRight.x - bottomLeft.x);
-      const height = Math.abs(topRight.y - bottomLeft.y);
+      const fovWidth = canvasSize.width * scale.x;
+      const fovHeight = canvasSize.height * scale.y;
 
-      const clampedPosition = clampPositionToArea(
-        worldPt,
-        { width, height },
-        visSize
+      // Create box to clamp target camera position, to disallow panning outside of visualization bounds
+      const clampingBox = Box.empty(new Vector3(0, 0, position.z)).expandBySize(
+        Math.max(visSize.width - fovWidth, 0),
+        Math.max(visSize.height - fovHeight, 0)
       );
 
-      position.set(clampedPosition.x, clampedPosition.y, position.z);
-
+      const target = clampingBox.clampPoint(worldPt, new Vector3());
+      position.copy(target);
       camera.updateMatrixWorld();
+
       invalidate();
     },
-    [camera, visSize, invalidate]
+    [camera, visSize, canvasSize, invalidate]
+  );
+}
+
+export function useZoomOnBox() {
+  const { canvasSize } = useVisCanvasContext();
+
+  const camera = useThree((state) => state.camera);
+  const moveCameraTo = useMoveCameraTo();
+
+  return useCallback(
+    (zoomBox: Box) => {
+      const { width, height } = canvasSize;
+
+      // Update camera scale first (since `moveCameraTo` relies on camera scale)
+      const { width: zoomWidth, height: zoomHeight } = zoomBox.size;
+      camera.scale.set(zoomWidth / width, zoomHeight / height, 1);
+
+      // Then move camera position
+      moveCameraTo(zoomBox.center);
```

Comment:
As mentioned, I might remove `moveCameraTo` and just set the camera position directly in a future PR.